In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats

# Problem - 01

Assuming

null hypothesis Ho -> μ1 = μ2 (There is no difference in diameters of cutlets between two units)

alternate hypothesis Ha -> μ1 ≠ μ2 (There is significant difference in diameters of cutlets between two units) 2 Sample 2 Tail test applicable


In [8]:
cutlet = pd.read_csv('Cutlets.csv')
cutlet.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [9]:
# 2-sample 2-tail ttest:   
#stats.ttest_ind(array1,array2)     
# ind -> independent samples

In [13]:
stats.ttest_ind( cutlet["Unit A"] , cutlet["Unit B"]) [1]

0.4722394724599501

In [14]:
# as the p value is greater than 0.05
# we will go with the h0 
# Means , there is no any significant difference in the diameter of the cutlet between two units.

In [15]:
#hence failed to reject null hypothesis (Ho), accept Ho.

# Problem - 02


assuming Ho -> No variance in TAT

assuming Ha -> Variance in TAT

In [22]:
import pandas as pd
from scipy import stats
from scipy.stats import norm

In [23]:
labtat = pd.read_csv('LabTAT.csv')
labtat.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [24]:
# here,  h0 : Laboratory 1 = Laboratory 2 = Laboratory 3 = Laboratory 4
#        h1 : one of them are not equal
#        we have to perform the ANOVA TEST

In [25]:
# Anova ftest statistics: stats.f_oneway(column-1,column-2,column-3,column-4)

stats.f_oneway( labtat['Laboratory 1'], labtat['Laboratory 2'], labtat['Laboratory 3'], labtat['Laboratory 4'] )

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [26]:
# as the p value is less than 0.05
# we will go with the h1
# Means, there is difference in the average Turn Around Time (TAT) of reports of the laboratories.

As p>α, failed to reject Ho; accept Ho.



# Problem - 03

Ho -> all proportions are equal

Ha -> not all proportions are equal

In [29]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [30]:
BR = pd.read_csv('BuyerRatio.csv')
BR

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [31]:
table = BR.iloc[:,1:6]
table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [32]:
# here,  h0: all proportional are equal
#        h1: all proprotional are not equal
#        we will perform CHI SQUARE TEST

In [34]:
stats.chi2_contingency (table) [1]

0.6603094907091882

In [35]:
# as we can see the p value is greater than 0.05
# we will reject the h0
# means, all proportional are not equal

p_value=0.66031

on comparing it with α = 0.05

as p_value>α, failed to reject Ho; accept Ho.

# Problem - 04

assuming Ho -> the defective % does not vary by centre

assuming Ha -> the defective % does vary by centre

In [36]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [37]:
telecall = pd.read_csv('Costomer+OrderForm.csv')
telecall

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [38]:
telecall['Phillippines'].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [39]:
telecall['Indonesia'].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [40]:
telecall['Malta'].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [41]:
telecall['India'].value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [42]:
df = { 'Phillippines':[271, 29],
       'Indonesia': [267, 33] , 
       'Malta':[269, 31],
       'India':[280, 20]     }

In [44]:
tab = pd.DataFrame (df)
tab

Phillippines  Indonesia  Malta  India
0           271        267    269    280
1            29         33     31     20

In [45]:
stats.chi2_contingency(tab)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [46]:
# as the p value is 0.277 which is greate than 0.05
# we will go with the h0
# means, the defective %  varies by centre

p_value=0.2771

as p_value>α = 0.05

failed to reject null hypothesis; accept Ho.